### PR TITLE
Add support for AUTH LOGIN method

### PR DESCRIPTION
**Pull request by maxhq**

### CODE REVIEW

**Comment by olegwtf** [comment]
Tests passes, but I got this warning
```
Mojo::Reactor::EV: I/O watcher failed: Can't call method "remaining" on an undefined value at /mnt/d/repos/github/p5-Mojo-SMTP-Client/lib/Mojo/SMTP/Client.pm line 204.
```

Please fix this

File: lib/Mojo/SMTP/Client.pm [hunk 1]
```diff
@@ -326,17 +326,52 @@ sub _make_cmd_steps {
...
+					my $delay = shift;
+					$self->_write_cmd('AUTH LOGIN', CMD_AUTH);
+					$self->_read_response($delay->begin, 0);
+					$self->{expected_code} = CMD_MORE;
+				}),
+				$self->{resp_checker},
+				# send username
+				$self->_new_cmd(sub {
+					my $delay = shift;
+					$self->_write_cmd(''.b64_encode($cmd[$mi]->{login}, ''), CMD_AUTH);
```
Comment (olegwtf) [inline lib/Mojo/SMTP/Client.pm]: why you need this empty strings here?

File: lib/Mojo/SMTP/Client.pm [hunk 2]
```diff
@@ -326,17 +326,52 @@ sub _make_cmd_steps {
...
+					my $delay = shift;
+					$self->_write_cmd(''.b64_encode($cmd[$mi]->{password}, ''), CMD_AUTH);
+					$self->_read_response($delay->begin, !$prepend && $mi == $#cmd);
+					$self->{expected_code} = CMD_OK;
+				}),
+				$self->{resp_checker},
+				$set_auth_ok;
+			}
+			else {
+			    croak 'unrecognized auth method: ', $type;
```
Comment (olegwtf) [inline lib/Mojo/SMTP/Client.pm:375]: Better to use `Mojo::SMTP::Client::Exception` here

Comment (olegwtf) [reply lib/Mojo/SMTP/Client.pm:375]: However, just checked other code, croak is good here.

File: t/03_smtp_nb.t [hunk 3]
```diff
@@ -525,6 +525,57 @@ $loop->reactor->remove($sock);
 close $sock;
 kill 15, $pid;
 
+# 9 - AUTH LOGIN method
+my ($pid, $sock, $host, $port) = Utils::make_smtp_server();
```
Comment (olegwtf) [inline t/03_smtp_nb.t]: my is useless here and below, this variables were declared above

File: lib/Mojo/SMTP/Client.pm (hunk 1 above)
Comment (maxhq) [reply lib/Mojo/SMTP/Client.pm]: Left-over from an ealier faulty use of b64_encode, sorry.

File: t/03_smtp_nb.t (hunk 3 above)
Comment (maxhq) [reply t/03_smtp_nb.t]: Sorry again, I copy-n-pasted your first test to get mine working, but forget to remove `my`.

**Comment by maxhq** [comment]
Concerning the errors from `Mojo::Reactor::EV`:
I cannot reproduce them but I got some strange `Mojo::IOLoop` errors earlier that disappeared after updating `Mojolicious` to `7.43`.

**Comment by olegwtf** [comment]
Looks good now. Warning gone away after you removed extra `my`. Will make new  release now.

**Pull request merged**